javasphinx/compiler.py

```diff
@@ -255,7 +255,7 @@ def compile_type_document(self, imports_block, package, name, declaration):
         fields = filter(self.filter, declaration.fields)
         if fields:
             document.add_heading('Fields', '-')
-            fields.sort(key=lambda f: f.declarators[0].name)
+            list(fields).sort(key=lambda f: f.declarators[0].name)
```

Comment:
Unfortunately this won't sort the contents of `fields`. `list(fields).sort(...)` creates and then sorts a _copy_ of the list, but leaves the original iterator returned by `filter` unmodified. I think we should instead wrap the `filter` call on line 255 with `list(...)`. This will also fixes an issue with the conditional on line 256. In 2.x `filter` returns a list, so the conditional will fail if the list is empty. In 3.x `filter` returns an iterator which always evaluates as true.